cli/internal/lockfile/pnpm_lockfile.go

```diff
@@ -0,0 +1,321 @@
+package lockfile
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"text/template"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+const pnpmLockfileTemplate = `lockfileVersion: {{ .Version }}
+
+importers:
+{{ range $key, $val := .Importers }}
+  {{ $key }}:
+{{ displayProjectSnapshot $val }}
+{{ end }}
+packages:
+{{ range $key, $val :=  .Packages }}
+  {{ $key }}:
+{{ displayPackageSnapshot $val }}
+{{ end }}{{ if (eq .Version 5.4) }}
+{{ end }}`
```

Comment:
A `pnpm` lockfile is 100% yaml. Is there a particular reason you construct it via a template instead of going all the way to yaml and then serializing it?

Reply:
The go yaml package explicitly doesn't support controlling whitespace: https://github.com/go-yaml/yaml/issues/627
Having encoding/decoding to preserve the file was really helpful for development when diffing the original and pruned lockfile. If sacrificing that for a straightforward yaml serialization is desired we can do that. I'll quick double check that pnpm doesn't throw if parsing these formatting differences.

Reply:
If `pnpm` can support arbitrary whitespace differences this may be a _feature_, not a bug. Non-meaningful whitespace inclusion would unnecessarily break a hash.